plugins/in_tail/tail_file.c

```diff
@@ -1448,6 +1492,46 @@ void flb_tail_file_remove(struct flb_tail_file *file)
     flb_plg_debug(ctx->ins, "inode=%"PRIu64" removing file name %s",
                   file->inode, file->name);
 
+    if (file->buf_len > 0) {
+        if (file->decompression_context == NULL) {
+            /*
+             * If there is data in the buffer, it means it was not processed.
+             * We must rewind the offset to ensure this data is re-read on restart.
+             */
+            int64_t old_offset = file->offset;
+
+            if (file->offset > file->buf_len) {
+                file->offset -= file->buf_len;
+            } else {
+                file->offset = 0;
+            }
```

Comment:
We need to put a newline before else statements.
In this case, we need to modify like:

```suggestion
            }
            else {
                file->offset = 0;
Comment view
            }
```

Reply:
Thanks for catching this! I've applied your suggestion and updated the code to put `else` on a new line.